packages/headless/src/controllers/product-listing/range-facet/date-facet/headless-produc-listing-date-facet.test.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  DateFacet,
+  buildDateFacet,
+  DateFacetOptions,
+} from './headless-product-listing-date-facet';
+import {
+  MockProductListingEngine,
+  buildMockProductListingEngine,
+} from '../../../../test/mock-engine';
+import {buildMockDateFacetValue} from '../../../../test/mock-date-facet-value';
+import {buildMockDateFacetRequest} from '../../../../test/mock-date-facet-request';
+import {configuration, dateFacetSet, search} from '../../../../app/reducers';
+import {DateFacetValue} from '../../../../features/facets/range-facets/date-facet-set/interfaces/response';
+import {ProductListingAppState} from '../../../../state/product-listing-app-state';
+import {buildMockProductListingState} from '../../../../test/mock-product-listing-state';
+import {fetchProductListing} from '../../../../features/product-listing/product-listing-actions';
+
+describe('date facet', () => {
+  const facetId = '1';
+  let options: DateFacetOptions;
+  let state: ProductListingAppState;
+  let engine: MockProductListingEngine;
+  let dateFacet: DateFacet;
+
+  function initDateFacet() {
+    engine = buildMockProductListingEngine({state});
+    dateFacet = buildDateFacet(engine, {options});
+  }
+
+  beforeEach(() => {
+    options = {
+      facetId,
+      field: 'created',
+      generateAutomaticRanges: true,
+    };
+
+    state = buildMockProductListingState();
+    state.dateFacetSet[facetId] = buildMockDateFacetRequest();
+
+    initDateFacet();
+  });
+
+  it('it adds the correct reducers to engine', () => {
+    expect(engine.addReducers).toHaveBeenCalledWith({
+      dateFacetSet,
+      configuration,
+      search,
+    });
+  });
+
+  describe('#toggleSelect', () => {
+    it('dispatches #fetchProductListing', () => {
+      const value = buildMockDateFacetValue();
+      dateFacet.toggleSelect(value);
+
+      const action = engine.actions.find(
+        (a) => a.type === fetchProductListing.pending.type
+      );
+      expect(action).toBeTruthy();
```

Comment:
`toBeDefined`? I dunno, its not exactly a boolean(ish) value, so truthy seems less obvious